packages/compass-data-modeling/src/utils/utils.ts

```diff
@@ -1,8 +1,56 @@
-import type { FieldPath } from '../services/data-model-storage';
+import type { FieldPath, Relationship } from '../services/data-model-storage';
 
 export function areFieldPathsEqual(
   fieldA: FieldPath,
   fieldB: FieldPath
 ): boolean {
   return JSON.stringify(fieldA) === JSON.stringify(fieldB);
 }
+
+export function isSameFieldOrAncestor(
+  ancestor: FieldPath,
+  child: FieldPath
+): boolean {
+  if (ancestor.length === child.length)
+    return areFieldPathsEqual(ancestor, child);
+  if (ancestor.length > child.length) return false;
+  // ignore the last character - closing bracket
+  const ancestorPath = JSON.stringify(ancestor).slice(0, -1);
+  const beginningOfChildPath = JSON.stringify(child).slice(
+    0,
+    ancestorPath.length
+  );
+  return ancestorPath === beginningOfChildPath;
```

Comment:
This is a bit hard to read, we're dealing with array of strings and they are easy to compare with conventional methods without serialization. Is this just checking that all of the items in array 1 equal the same items in array 2?

```suggestion
  return ancestor.every((pathPart, index) => pathPart === child[index]);
```

Reply:
🤦